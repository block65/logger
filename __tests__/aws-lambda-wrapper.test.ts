import {
  afterEach,
  beforeEach,
  describe,
  expect,
  jest,
  test,
} from '@jest/globals';

describe('AWS', () => {
  const initialEnv = process.env;
  beforeEach(() => {
    // jest.clearAllMocks();
    jest.useFakeTimers({ now: new Date('2009-02-13T23:31:30.000Z') });
  });

  afterEach(() => {
    process.env = { ...initialEnv };
    jest.useRealTimers();
  });

  describe('Lambda', () => {
    test('Logger + contextId', async () => {
      process.env.AWS_LAMBDA_FUNCTION_VERSION = '$LATEST';

      const { createAutoConfiguredLoggerWithWaitableMock } = await import(
        './helpers.js'
      );
      const { withLambdaLoggerContextWrapper } = await import(
        '../lib/lambda.js'
      );

      const [logger, callback, errback] =
        createAutoConfiguredLoggerWithWaitableMock();

      const result = await withLambdaLoggerContextWrapper(
        logger,
        {
          awsRequestId: '000fake-0000-000request-000-id',
          callbackWaitsForEmptyEventLoop: false,
          functionName: 'fakeFunctionName',
          functionVersion: '$LATEST',
          invokedFunctionArn: 'arn::::',
          logGroupName: 'fakeLogGroup',
          logStreamName: 'fakeLogStream',
          memoryLimitInMB: '128',
        },
        async () => {
          logger.warn('hello');
          logger.error(new Error('fake'));
          return {
            somethingUseful: true,
          };
        },
      );

      expect(result).toMatchSnapshot();

      await callback.waitUntilCalled();

      expect(callback).toBeCalledTimes(2);
      expect(callback.mock.calls).toMatchSnapshot();
      expect(errback).not.toBeCalled();
    });
  });
});
