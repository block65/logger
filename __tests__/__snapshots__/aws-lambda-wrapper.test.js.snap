// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AWS Lambda Logger + contextId 1`] = `
Object {
  "somethingUseful": true,
}
`;

exports[`AWS Lambda Logger + contextId 2`] = `
Array [
  Array [
    "2009-02-13T23:31:30.000Z	000fake-0000-000request-000-id	WARN	hello	{\\"level\\":40}
",
  ],
  Array [
    "2009-02-13T23:31:30.000Z	000fake-0000-000request-000-id	ERROR	fake	{\\"level\\":50,\\"err\\":{\\"name\\":\\"Error\\",\\"message\\":\\"fake\\",\\"stack\\":\\"Error: fake\\\\n    at fn (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/lib/lambda.js:12:16\\\\n    at AsyncLocalStorage.run (node:async_hooks:320:14)\\"}}
",
  ],
]
`;
